model/src/main/proto/arguments.proto

```diff
@@ -326,3 +327,516 @@ message SurveyActivityParams {
+  bool is_multipane = 1;
+}
+
+// Represents the parameters needed to create OptionsActivity.
+message OptionsActivityParams {
+  // Represents whether the user navigated to the OptionsActivity through the navigation drawer or not.
+  bool is_from_navigation_drawer = 1;
+}
+
+// The bundle of properties that are saved upon configuration changes in OptionsActivity.
+message OptionsActivityStateBundle {
+  // Title for loaded fragment.
+  string selected_options_title = 1;
+
+  // Stores the identifier of the currently selected fragment.
+  string selected_fragment = 2;
+}
+
+// Arguments required when creating an OptionsFragment.
+message OptionsFragmentArguments {
+  // The value indicates whether the layout is multipane or not.
+  bool is_multipane = 1;
+
+  // Indicates whether to load the default fragment or not.
+  bool is_first_open = 2;
+
+  // Stores the identifier of the currently selected fragment.
+  string selected_fragment = 3;
+}
+
+// Arguments required when creating a HintsAndSolutionDialogFragment.
+message HintsAndSolutionDialogFragmentArguments {
+  // Id Used in ExplorationController/QuestionAssessmentProgressController to get current state data.
+  string id_argument = 1;
+
+  // The [State] being viewed by the learner.
+  State state = 2;
+
+  // [HelpIndex] corresponding to the current hints/solution configuration.
+  HelpIndex help_index = 3;
+
+  // [WrittenTranslationContext] needed to translate the hints/solution.
+  WrittenTranslationContext written_translation_context = 4;
+}
+
+// The bundle of properties that are saved upon configuration changes in HintsAndSolutionDialogFragment.
+message HintsAndSolutionDialogFragmentStateBundle {
+  // List of expanded items.
+  repeated int32 current_expanded_items = 1;
+
+  // Index of hint.
+  int32 hint_index = 2;
+
+  // Indicates whether hint is revealed or not.
+  bool is_hint_revealed = 3;
+
+  // Solution index.
+  int32 solution_index = 4;
+
+  // Indicates whether solution is revealed or not.
+  bool is_solution_revealed = 5;
+}
+
+// Arguments required when creating a LanguageDialogFragment.
+message LanguageDialogFragmentArguments {
+  // List of strings containing languages.
+  repeated string languages = 1;
+
+  // Selected index from language list.
+  int32 selected_index = 2;
+}
+
+// Represents the parameters needed to create StateFragmentTestActivity.
+message StateFragmentTestActivityParams {
+
+  // The ID of the current profile.
+  int32 internal_profile_id = 1;
+
+  // The ID of the topic to which the triggering exploration belongs.
+  string topic_id = 2;
+
+  // The ID of the story to which the triggering exploration belongs.
+  string story_id = 3;
+
+  // The ID of the triggering exploration.
+  string exploration_id = 4;
+
+  // Indicates whether to save partial progress or not.
+  bool should_save_partial_progress = 5;
+}
+
+// Arguments required when creating a StateFragment.
+message StateFragmentArguments {
+
```

Comment:
Nit here & above & elsewhere: please remove extra starting newline for formatting consistency.

Reply:
Changes made.